nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -411,10 +391,44 @@ private Description checkParamOverriding(
         return Description.NO_MATCH;
     }
 
+    private Description checkReturnExpression(
+            Tree tree,
+            ExpressionTree retExpr,
+            Symbol.MethodSymbol methodSymbol,
+            VisitorState state) {
+        Type returnType = methodSymbol.getReturnType();
+        if (returnType.isPrimitive()) {
+            // check for unboxing
+            return doUnboxingCheck(state, retExpr);
+        }
+        if (returnType.toString().equals("java.lang.Void")) {
+            return Description.NO_MATCH;
+        }
+        if (fromUnannotatedPackage(methodSymbol)
+                || TrustingNullnessAnalysis.hasNullableAnnotation(methodSymbol)) {
+            return Description.NO_MATCH;
+        }
+        if (mayBeNullExpr(state, retExpr)) {
+            return createErrorDescription(
+                    tree,
+                    "returning @Nullable expression from method with @NonNull return type" + methodSymbol,
```

Comment:
Nit: `type "` instead of `type"`

Reply:
Oh, whoops, that methodSymbol shouldn't even be there, that was for debugging. Good catch.